### PR TITLE
docs: update prop labels for maxLength and maxCharacters(#2743)

### LINKED ISSUE #1726: Internal issue 2743 - IcTextField documentation is inaccurate
The docs for two of the IcTextField props could be improved for the following 2 props:

maxCharacters - The maximum number of characters that can be entered in the field.

maxLength - The label for maximum number of characters that can be entered in the field.

For maxCharacters, this is actually a “hard limit” on the number of characters that can be entered into the field. It shows a warning state if a user tries to enter too many characters.

Whereas for maxLength, this is a character limit but it allows the user to enter more characters if they wish. If they do so it shows an error state.

It would be helpful if the docs could be updated at some point to reflect this.

### PR DESCRIPTION
**Pull request by evenstensberg**
## Summary of the changes
Changes wording of the two props `maxLength`and `maxCharacters`. Let me know if you want me to change the sentences to a more detailed one.

As far as I understood, maxLength will show an error if bounds are reached, while maxCharacters will show a warning. 

## Related issue
Closes #1726

### General 

- [x] Changes to docs package checked and committed.
- [ ] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

**Comment by GCHQ-Developer-847** [comment]
Hi @evenstensberg, thanks for your contribution!
 
I feel like the behaviour and difference between the two props could be clearer (see the [maxLength](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-text-field--max-length) and [maxCharacters](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-text-field--maximum-and-minimum-characters) stories). I personally think the `maxCharacters` description was already clear enough - maybe just adding a few more words onto the end about the warning would work? And I think the two main things we'd want to include in the `maxLength` description should essentially be that it displays a character count, and that it shows an error when the value is exceeded. This is all just my opinion though so happy if we want to go with something different!
 
Also, the prop descriptions in the `docs.json` file are automatically generated based on the comments above each prop in the file for each component (e.g. `ic-text-field.tsx`). Please could you change the prop descriptions in there instead, and then running "[npm run build]([https://github.com/mi6/ic-ui-kit/blob/fc202d5fde5e93c4e744747d115d83d09901cf79/README.md?plain=1#L43)"](https://github.com/mi6/ic-ui-kit/blob/fc202d5fde5e93c4e744747d115d83d09901cf79/README.md?plain=1#L43)%22) will update `docs.json`.
 
Another thing - the tasks in the PR description need to be completed for the checks to pass. Any that aren't applicable you can mark as done.
 
Thanks :)

**Comment by GCHQ-Developer-847** [comment]
Hi @evenstensberg, thanks for your contribution!
 
I feel like the behaviour and difference between the two props could be clearer (see the [maxLength](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-text-field--max-length) and [maxCharacters](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-text-field--maximum-and-minimum-characters) stories). I personally think the `maxCharacters` description was already clear enough - maybe just adding a few more words onto the end about the warning would work? And I think the two main things we'd want to include in the `maxLength` description should essentially be that it displays a character count, and that it shows an error when the value is exceeded. This is all just my opinion though so happy to hear other thoughts!
 
Also, the prop descriptions in the `docs.json` file are automatically generated based on the comments above each prop in the file for each component (e.g. `ic-text-field.tsx`). Please could you change the prop descriptions in there instead, and then running "[npm run build](https://github.com/mi6/ic-ui-kit/blob/fc202d5fde5e93c4e744747d115d83d09901cf79/README.md?plain=1#L43)" will update `docs.json`.
 
Another thing - the tasks in the PR description need to be completed for the checks to pass. Any that aren't applicable you can mark as done.
 
Thanks :)

File: packages/web-components/src/components/ic-text-field/ic-text-field.tsx [hunk 1]
```diff
@@ -162,12 +162,12 @@ export class TextField {
...
 
   /**
-   * The maximum number of characters that can be entered in the field.
+   * The maximum number of characters that can be entered in the field. Will emit an error if too many characters are entered. 
    */
   @Prop() maxCharacters: number = 0;
 
   /**
-   * The label for maximum number of characters that can be entered in the field.
+   * Count of characters in the field. Will emit an error if the bound is reached.
```
Comment (GCHQ-Developer-530) [inline packages/web-components/src/components/ic-text-field/ic-text-field.tsx:170]:
```suggestion
   * The count of characters in the field. Will emit an error if the bound is reached.
```

**Comment by GCHQ-Developer-530** [comment]
This change is failing the prettier check in ic-text-field.tsx, please could you run `npm run prettier:fix` and commit the update to that file?

File: packages/web-components/src/components/ic-text-field/ic-text-field.tsx [hunk 2]
```diff
@@ -162,12 +162,12 @@ export class TextField {
   @Prop() max: string | number = undefined;
 
   /**
-   * The maximum number of characters that can be entered in the field.
+   * The maximum number of characters that can be entered in the field. Will emit an error if too many characters are entered.
```
Comment (GCHQ-Developer-847) [inline packages/web-components/src/components/ic-text-field/ic-text-field.tsx:165]: I feel like "display" might be more appropriate as "emit" is generally associated with events
```suggestion
   * The maximum number of characters that can be entered in the field. Will display an error if too many characters are entered.
```

File: packages/web-components/src/components/ic-text-field/ic-text-field.tsx [hunk 3]
```diff
@@ -162,12 +162,12 @@ export class TextField {
...
    */
   @Prop() maxCharacters: number = 0;
 
   /**
-   * The label for maximum number of characters that can be entered in the field.
+   * The count of characters in the field. Will emit an error if the bound is reached.
```
Comment (GCHQ-Developer-847) [inline packages/web-components/src/components/ic-text-field/ic-text-field.tsx:170]:
```suggestion
   * The count of characters in the field. Will display a warning if the bound is reached.
```

**Review by GCHQ-Developer-847 — COMMENTED** [review]
Just a couple of small things

Also the auto-generated files will need updating by doing an `npm run build` again

**Comment by evenstensberg** [comment]
@GCHQ-Developer-847 Fixed now! I had to reset the branch and re-add the changes because the rebase I did, failed.

**Review by GCHQ-Developer-847 — APPROVED** [review]
Looks great! Thanks for making all the changes :)

**Comment by evenstensberg** [comment]
And thanks for reviewing!

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**